ccl/multiregionccl: unskip test multi region dd parent test

kvoli opened the pull request:
`TestMultiRegionDataDriven` was skipped in https://github.com/cockroachdb/cockroach/pull/108107 due to finish
(tracing span) being called twice in raft reproposals https://github.com/cockroachdb/cockroach/issues/107521, which
 https://github.com/cockroachdb/cockroach/pull/108775 fixed.

Unskip the top level `TestMultiRegionDataDriven` test. Note
`/secondary_region` is still skipped due to a known allocator bug.

Informs: https://github.com/cockroachdb/cockroach/issues/98020
Epic: none
Release note: None

kvoli committed:
ccl/multiregionccl: unskip test multi region dd parent test

`TestMultiRegionDataDriven` was skipped in #108107 due to finish
(tracing span) being called twice in raft reproposals #107521, which
 #108775 fixed.

Unskip the top level `TestMultiRegionDataDriven` test. Note
`/secondary_region` is still skipped due to a known allocator bug.

Informs: #98020
Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110063)
<!-- Reviewable:end -->

kvoli committed:
ccl/multiregionccl: fix setting buglet in mr test

Update the cluster setting statement to correctly surround the value
duration in quotations.`5m` to `'5m'`. This was fixed in backports to
release-23.1 (#110052) and release-22.2 (#110053). However, this was not
caught on master due to `TestMultiRegionDataDriven` being skipped.

Informs: #98020
Release note: None

kvoli commented:
TYFTR

bors r=rafiss
